File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/LoggedModelInfo.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * Details of the LoggedModel Info.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=LoggedModelInfo.Builder.class)

public final class LoggedModelInfo  {
    @Deprecated
    @java.beans.ConstructorProperties({"modelId", "experimentId", "name", "creationTimestampMs", "lastUpdatedTimestampMs", "artifactUri", "status", "modelType", "sourceRunId", "tags"})
    public LoggedModelInfo(String modelId, String experimentId, String name, Long creationTimestampMs, Long lastUpdatedTimestampMs, String artifactUri, String status, String modelType, String sourceRunId, java.util.List<LoggedModelTag> tags) {
        super();
        this.modelId = modelId;
        this.experimentId = experimentId;
        this.name = name;
        this.creationTimestampMs = creationTimestampMs;
        this.lastUpdatedTimestampMs = lastUpdatedTimestampMs;
        this.artifactUri = artifactUri;
        this.status = status;
        this.modelType = modelType;
        this.sourceRunId = sourceRunId;
        this.tags = tags;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * id of logged-model.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("model_id")
private String modelId;

        /**
         * id of logged-model.
         * @param modelId the value to set
         * @return this builder
         **/
        

public Builder modelId(String modelId) {
    this.modelId = modelId;
    return this;
}
            /**
     * Unique identifier for the experiment.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("experiment_id")
private String experimentId;

        /**
         * Unique identifier for the experiment.
         * @param experimentId the value to set
         * @return this builder
         **/
        

public Builder experimentId(String experimentId) {
    this.experimentId = experimentId;
    return this;
}
            /**
     * name of logged-model.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("name")
private String name;

        /**
         * name of logged-model.
         * @param name the value to set
         * @return this builder
         **/
        

public Builder name(String name) {
    this.name = name;
    return this;
}
            /**
     * Unix timestamp in milliseconds when the logged-model was created.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("creation_timestamp_ms")
private Long creationTimestampMs;

        /**
         * Unix timestamp in milliseconds when the logged-model was created.
         * @param creationTimestampMs the value to set
         * @return this builder
         **/
        

public Builder creationTimestampMs(Long creationTimestampMs) {
    this.creationTimestampMs = creationTimestampMs;
    return this;
}
            /**
     * Unix timestamp in milliseconds when the logged-model was last updated.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("last_updated_timestamp_ms")
private Long lastUpdatedTimestampMs;

        /**
         * Unix timestamp in milliseconds when the logged-model was last updated.
         * @param lastUpdatedTimestampMs the value to set
         * @return this builder
         **/
        

public Builder lastUpdatedTimestampMs(Long lastUpdatedTimestampMs) {
    this.lastUpdatedTimestampMs = lastUpdatedTimestampMs;
    return this;
}
            /**
     * artifact_uri.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("artifact_uri")
private String artifactUri;

        /**
         * artifact_uri.
         * @param artifactUri the value to set
         * @return this builder
         **/
        

public Builder artifactUri(String artifactUri) {
    this.artifactUri = artifactUri;
    return this;
}
            /**
     * status of logged-model.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("status")
private String status;

        /**
         * status of logged-model.
         * @param status the value to set
         * @return this builder
         **/
        

public Builder status(String status) {
    this.status = status;
    return this;
}
            /**
     * model_type of logged-model.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("model_type")
private String modelType;

        /**
         * model_type of logged-model.
         * @param modelType the value to set
         * @return this builder
         **/
        

public Builder modelType(String modelType) {
    this.modelType = modelType;
    return this;
}
            /**
     * source_run_id of logged-model.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("source_run_id")
private String sourceRunId;

        /**
         * source_run_id of logged-model.
         * @param sourceRunId the value to set
         * @return this builder
         **/
        

public Builder sourceRunId(String sourceRunId) {
    this.sourceRunId = sourceRunId;
    return this;
}
            /**
     * tags of logged-model
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("tags")
private java.util.List<LoggedModelTag> tags;

        /**
         * tags of logged-model
         * @param tags the value to set
         * @return this builder
         **/
        

public Builder tags(java.util.List<LoggedModelTag> tags) {
    this.tags = tags;
    return this;
}


        public LoggedModelInfo build() {
            LoggedModelInfo model = new LoggedModelInfo(this.modelId
                , this.experimentId
                , this.name
                , this.creationTimestampMs
                , this.lastUpdatedTimestampMs
                , this.artifactUri
                , this.status
                , this.modelType
                , this.sourceRunId
                , this.tags);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(LoggedModelInfo model) {
                this.modelId(model.getModelId());
    this.experimentId(model.getExperimentId());
    this.name(model.getName());
    this.creationTimestampMs(model.getCreationTimestampMs());
    this.lastUpdatedTimestampMs(model.getLastUpdatedTimestampMs());
    this.artifactUri(model.getArtifactUri());
    this.status(model.getStatus());
    this.modelType(model.getModelType());
    this.sourceRunId(model.getSourceRunId());
    this.tags(model.getTags());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * id of logged-model.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("model_id")
    private final String modelId;

        /**
     * id of logged-model.
     * @return the value
     **/
    
    public String getModelId() {
        return modelId;
    }


        /**
     * Unique identifier for the experiment.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("experiment_id")
    private final String experimentId;

        /**
     * Unique identifier for the experiment.
     * @return the value
     **/
    
    public String getExperimentId() {
        return experimentId;
    }


        /**
     * name of logged-model.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("name")
    private final String name;

        /**
     * name of logged-model.
     * @return the value
     **/
    
    public String getName() {
        return name;
    }


        /**
     * Unix timestamp in milliseconds when the logged-model was created.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("creation_timestamp_ms")
    private final Long creationTimestampMs;

        /**
     * Unix timestamp in milliseconds when the logged-model was created.
     * @return the value
     **/
    
    public Long getCreationTimestampMs() {
        return creationTimestampMs;
    }


        /**
     * Unix timestamp in milliseconds when the logged-model was last updated.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("last_updated_timestamp_ms")
    private final Long lastUpdatedTimestampMs;

        /**
     * Unix timestamp in milliseconds when the logged-model was last updated.
     * @return the value
     **/
    
    public Long getLastUpdatedTimestampMs() {
        return lastUpdatedTimestampMs;
    }


        /**
     * artifact_uri.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("artifact_uri")
    private final String artifactUri;

        /**
     * artifact_uri.
     * @return the value
     **/
    
    public String getArtifactUri() {
        return artifactUri;
    }


        /**
     * status of logged-model.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("status")
    private final String status;

        /**
     * status of logged-model.
     * @return the value
     **/
    
    public String getStatus() {
        return status;
    }


        /**
     * model_type of logged-model.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("model_type")
    private final String modelType;

        /**
     * model_type of logged-model.
     * @return the value
     **/
    
    public String getModelType() {
        return modelType;
    }


        /**
     * source_run_id of logged-model.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("source_run_id")
    private final String sourceRunId;

        /**
     * source_run_id of logged-model.
     * @return the value
     **/
    
    public String getSourceRunId() {
        return sourceRunId;
    }


        /**
     * tags of logged-model
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("tags")
    private final java.util.List<LoggedModelTag> tags;

        /**
     * tags of logged-model
     * @return the value
     **/
    
    public java.util.List<LoggedModelTag> getTags() {
        return tags;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("LoggedModelInfo(");
        sb.append("modelId=").append(String.valueOf(this.modelId));
        sb.append(", experimentId=").append(String.valueOf(this.experimentId));
        sb.append(", name=").append(String.valueOf(this.name));
        sb.append(", creationTimestampMs=").append(String.valueOf(this.creationTimestampMs));
        sb.append(", lastUpdatedTimestampMs=").append(String.valueOf(this.lastUpdatedTimestampMs));
        sb.append(", artifactUri=").append(String.valueOf(this.artifactUri));
        sb.append(", status=").append(String.valueOf(this.status));
        sb.append(", modelType=").append(String.valueOf(this.modelType));
        sb.append(", sourceRunId=").append(String.valueOf(this.sourceRunId));
        sb.append(", tags=").append(String.valueOf(this.tags));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof LoggedModelInfo)) {
            return false;
        }

        LoggedModelInfo other = (LoggedModelInfo) o;
        return java.util.Objects.equals(this.modelId, other.modelId) &&
            java.util.Objects.equals(this.experimentId, other.experimentId) &&
            java.util.Objects.equals(this.name, other.name) &&
            java.util.Objects.equals(this.creationTimestampMs, other.creationTimestampMs) &&
            java.util.Objects.equals(this.lastUpdatedTimestampMs, other.lastUpdatedTimestampMs) &&
            java.util.Objects.equals(this.artifactUri, other.artifactUri) &&
            java.util.Objects.equals(this.status, other.status) &&
            java.util.Objects.equals(this.modelType, other.modelType) &&
            java.util.Objects.equals(this.sourceRunId, other.sourceRunId) &&
            java.util.Objects.equals(this.tags, other.tags);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.modelId == null ? 43 : this.modelId.hashCode());
        result = (result * PRIME) + (this.experimentId == null ? 43 : this.experimentId.hashCode());
        result = (result * PRIME) + (this.name == null ? 43 : this.name.hashCode());
        result = (result * PRIME) + (this.creationTimestampMs == null ? 43 : this.creationTimestampMs.hashCode());
        result = (result * PRIME) + (this.lastUpdatedTimestampMs == null ? 43 : this.lastUpdatedTimestampMs.hashCode());
        result = (result * PRIME) + (this.artifactUri == null ? 43 : this.artifactUri.hashCode());
        result = (result * PRIME) + (this.status == null ? 43 : this.status.hashCode());
        result = (result * PRIME) + (this.modelType == null ? 43 : this.modelType.hashCode());
        result = (result * PRIME) + (this.sourceRunId == null ? 43 : this.sourceRunId.hashCode());
        result = (result * PRIME) + (this.tags == null ? 43 : this.tags.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/logged_model_info.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class LoggedModelInfo(object):
    """
    Details of the LoggedModel Info.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new LoggedModelInfo object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param model_id:
            The value to assign to the model_id property of this LoggedModelInfo.
        :type model_id: str

        :param experiment_id:
            The value to assign to the experiment_id property of this LoggedModelInfo.
        :type experiment_id: str

        :param name:
            The value to assign to the name property of this LoggedModelInfo.
        :type name: str

        :param creation_timestamp_ms:
            The value to assign to the creation_timestamp_ms property of this LoggedModelInfo.
        :type creation_timestamp_ms: int

        :param last_updated_timestamp_ms:
            The value to assign to the last_updated_timestamp_ms property of this LoggedModelInfo.
        :type last_updated_timestamp_ms: int

        :param artifact_uri:
            The value to assign to the artifact_uri property of this LoggedModelInfo.
        :type artifact_uri: str

        :param status:
            The value to assign to the status property of this LoggedModelInfo.
        :type status: str

        :param model_type:
            The value to assign to the model_type property of this LoggedModelInfo.
        :type model_type: str

        :param source_run_id:
            The value to assign to the source_run_id property of this LoggedModelInfo.
        :type source_run_id: str

        :param tags:
            The value to assign to the tags property of this LoggedModelInfo.
        :type tags: list[oci.aidataplatform_dp.models.LoggedModelTag]

        """
        self.swagger_types = {
            'model_id': 'str',
            'experiment_id': 'str',
            'name': 'str',
            'creation_timestamp_ms': 'int',
            'last_updated_timestamp_ms': 'int',
            'artifact_uri': 'str',
            'status': 'str',
            'model_type': 'str',
            'source_run_id': 'str',
            'tags': 'list[LoggedModelTag]'
        }

        self.attribute_map = {
            'model_id': 'model_id',
            'experiment_id': 'experiment_id',
            'name': 'name',
            'creation_timestamp_ms': 'creation_timestamp_ms',
            'last_updated_timestamp_ms': 'last_updated_timestamp_ms',
            'artifact_uri': 'artifact_uri',
            'status': 'status',
            'model_type': 'model_type',
            'source_run_id': 'source_run_id',
            'tags': 'tags'
        }

        self._model_id = None
        self._experiment_id = None
        self._name = None
        self._creation_timestamp_ms = None
        self._last_updated_timestamp_ms = None
        self._artifact_uri = None
        self._status = None
        self._model_type = None
        self._source_run_id = None
        self._tags = None

    @property
    def model_id(self):
        """
        Gets the model_id of this LoggedModelInfo.
        id of logged-model.


        :return: The model_id of this LoggedModelInfo.
        :rtype: str
        """
        return self._model_id

    @model_id.setter
    def model_id(self, model_id):
        """
        Sets the model_id of this LoggedModelInfo.
        id of logged-model.


        :param model_id: The model_id of this LoggedModelInfo.
        :type: str
        """
        self._model_id = model_id

    @property
    def experiment_id(self):
        """
        Gets the experiment_id of this LoggedModelInfo.
        Unique identifier for the experiment.


        :return: The experiment_id of this LoggedModelInfo.
        :rtype: str
        """
        return self._experiment_id

    @experiment_id.setter
    def experiment_id(self, experiment_id):
        """
        Sets the experiment_id of this LoggedModelInfo.
        Unique identifier for the experiment.


        :param experiment_id: The experiment_id of this LoggedModelInfo.
        :type: str
        """
        self._experiment_id = experiment_id

    @property
    def name(self):
        """
        Gets the name of this LoggedModelInfo.
        name of logged-model.


        :return: The name of this LoggedModelInfo.
        :rtype: str
        """
        return self._name

    @name.setter
    def name(self, name):
        """
        Sets the name of this LoggedModelInfo.
        name of logged-model.


        :param name: The name of this LoggedModelInfo.
        :type: str
        """
        self._name = name

    @property
    def creation_timestamp_ms(self):
        """
        Gets the creation_timestamp_ms of this LoggedModelInfo.
        Unix timestamp in milliseconds when the logged-model was created.


        :return: The creation_timestamp_ms of this LoggedModelInfo.
        :rtype: int
        """
        return self._creation_timestamp_ms

    @creation_timestamp_ms.setter
    def creation_timestamp_ms(self, creation_timestamp_ms):
        """
        Sets the creation_timestamp_ms of this LoggedModelInfo.
        Unix timestamp in milliseconds when the logged-model was created.


        :param creation_timestamp_ms: The creation_timestamp_ms of this LoggedModelInfo.
        :type: int
        """
        self._creation_timestamp_ms = creation_timestamp_ms

    @property
    def last_updated_timestamp_ms(self):
        """
        Gets the last_updated_timestamp_ms of this LoggedModelInfo.
        Unix timestamp in milliseconds when the logged-model was last updated.


        :return: The last_updated_timestamp_ms of this LoggedModelInfo.
        :rtype: int
        """
        return self._last_updated_timestamp_ms

    @last_updated_timestamp_ms.setter
    def last_updated_timestamp_ms(self, last_updated_timestamp_ms):
        """
        Sets the last_updated_timestamp_ms of this LoggedModelInfo.
        Unix timestamp in milliseconds when the logged-model was last updated.


        :param last_updated_timestamp_ms: The last_updated_timestamp_ms of this LoggedModelInfo.
        :type: int
        """
        self._last_updated_timestamp_ms = last_updated_timestamp_ms

    @property
    def artifact_uri(self):
        """
        Gets the artifact_uri of this LoggedModelInfo.
        artifact_uri.


        :return: The artifact_uri of this LoggedModelInfo.
        :rtype: str
        """
        return self._artifact_uri

    @artifact_uri.setter
    def artifact_uri(self, artifact_uri):
        """
        Sets the artifact_uri of this LoggedModelInfo.
        artifact_uri.


        :param artifact_uri: The artifact_uri of this LoggedModelInfo.
        :type: str
        """
        self._artifact_uri = artifact_uri

    @property
    def status(self):
        """
        Gets the status of this LoggedModelInfo.
        status of logged-model.


        :return: The status of this LoggedModelInfo.
        :rtype: str
        """
        return self._status

    @status.setter
    def status(self, status):
        """
        Sets the status of this LoggedModelInfo.
        status of logged-model.


        :param status: The status of this LoggedModelInfo.
        :type: str
        """
        self._status = status

    @property
    def model_type(self):
        """
        Gets the model_type of this LoggedModelInfo.
        model_type of logged-model.


        :return: The model_type of this LoggedModelInfo.
        :rtype: str
        """
        return self._model_type

    @model_type.setter
    def model_type(self, model_type):
        """
        Sets the model_type of this LoggedModelInfo.
        model_type of logged-model.


        :param model_type: The model_type of this LoggedModelInfo.
        :type: str
        """
        self._model_type = model_type

    @property
    def source_run_id(self):
        """
        Gets the source_run_id of this LoggedModelInfo.
        source_run_id of logged-model.


        :return: The source_run_id of this LoggedModelInfo.
        :rtype: str
        """
        return self._source_run_id

    @source_run_id.setter
    def source_run_id(self, source_run_id):
        """
        Sets the source_run_id of this LoggedModelInfo.
        source_run_id of logged-model.


        :param source_run_id: The source_run_id of this LoggedModelInfo.
        :type: str
        """
        self._source_run_id = source_run_id

    @property
    def tags(self):
        """
        Gets the tags of this LoggedModelInfo.
        tags of logged-model


        :return: The tags of this LoggedModelInfo.
        :rtype: list[oci.aidataplatform_dp.models.LoggedModelTag]
        """
        return self._tags

    @tags.setter
    def tags(self, tags):
        """
        Sets the tags of this LoggedModelInfo.
        tags of logged-model


        :param tags: The tags of this LoggedModelInfo.
        :type: list[oci.aidataplatform_dp.models.LoggedModelTag]
        """
        self._tags = tags

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
